lib/src/Navigation.ts

```diff
@@ -16,6 +16,8 @@ import { ComponentEventsObserver } from './events/ComponentEventsObserver';
 import { TouchablePreview } from './adapters/TouchablePreview';
 import { LayoutRoot, Layout } from './interfaces/Layout';
 import { Options } from './interfaces/Options';
+import { Provider } from 'react-redux';
+import { Store as ReduxStore } from 'redux';
```

Comment:
I don't think so since these are used only as types so TS compiler will strip types away. If we wan't to be sure we could put these as dependencies and not devDependencies. 

Side note from me is that I think this library should have state management specific stuff in it. It should give some generic way of adding redux and other libraries with this library. But I can make other pr about that later if you guys want :)

Reply:
@henrikra I tried compiling your branch without redux dependencies - it does in fact break with a `Cannot find module 'redux'.` exceptions during tsc.

I this we should address redux issues in a separate pr, we tried to avoid coupling with redux as much as possible. Eventually we had to improve redux support and `registerComponentWithRedux` is the result. Perhaps we can/should pull it to a different library 🤔 

So to move forward, I suggest we leave redux issues for later and merge the rest of the PR. WDYT?